futures-util/src/try_future/mod.rs

```diff
@@ -501,4 +504,155 @@ pub trait TryFutureExt: TryFuture {
     {
         IntoFuture::new(self)
     }
+
+    /// Joins the result of two futures, waiting for them both to complete or
+    /// for one to produce an error.
+    ///
+    /// This function will return a new future which awaits both this and the
+    /// `other` future to complete. If successful, the returned future will
+    /// finish with a tuple of both results. If unsuccesful, it will complete
+    /// with the first error encountered.
+    ///
+    /// Note that this function consumes the receiving future and returns a
+    /// wrapped version of it.
+    ///
+    /// # Examples
+    ///
+    /// When used on multiple futures that reutrn `Ok`, `try_join` will return
+    /// `Ok` of a tuple of the values:
+    ///
+    /// ```
+    /// #![feature(async_await, await_macro, futures_api)]
+    /// # futures::executor::block_on(async {
+    /// use futures::future::{self, TryFutureExt};
+    ///
+    /// let a = future::ready(Ok::<i32, i32>(1));
+    /// let b = future::ready(Ok::<i32, i32>(2));
+    /// let pair = a.try_join(b);
+    ///
+    /// assert_eq!(await!(pair), Ok((1, 2)));
+    /// # });
+    /// ```
+    ///
+    /// If one of the futures resolves to an error, `try_join` will return
+    /// that error:
+    ///
+    /// ```
+    /// #![feature(async_await, await_macro, futures_api)]
+    /// # futures::executor::block_on(async {
+    /// use futures::future::{self, TryFutureExt};
+    ///
+    /// let a = future::ready(Ok::<i32, i32>(1));
+    /// let b = future::ready(Err::<i32, i32>(2));
+    /// let pair = a.try_join(b);
+    ///
+    /// assert_eq!(await!(pair), Err(2));
+    /// # });
+    /// ```
+    fn try_join<Fut2>(self, other: Fut2) -> TryJoin<Self, Fut2>
+    where
+        Fut2: TryFuture<Error = Self::Error>,
+        Self: Sized,
+    {
+        TryJoin::new(self, other)
+    }
+
+    /// Same as [`try_join`](TryFutureExt::try_join), but with more futures.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(async_await, await_macro, futures_api)]
+    /// # futures::executor::block_on(async {
+    /// use futures::future::{self, TryFutureExt};
+    ///
+    /// let a = future::ready(Ok::<i32, i32>(1));
+    /// let b = future::ready(Ok::<i32, i32>(2));
+    /// let c = future::ready(Err::<i32, i32>(3));
+    /// let tuple = a.try_join3(b, c);
+    ///
+    /// assert_eq!(await!(tuple), Err(3));
```

Comment:
I think it's better to show the success case like for `Join4` and `Join5`.

Reply:
Done.